packages/aragon-wrapper/src/index.js

```diff
@@ -828,6 +831,130 @@ export default class Aragon {
     this.appIdentifiers.connect()
   }
 
+  /**
+   * Initialise token manager observable.
+   *
+   * @return {void}
+   */
+  initTokenManagers () {
+    const tokenManagerID = '0x6b20a3010614eeebf2138ccec99f028a61c811b3b1a3343b6ff635985c75c91f'
+    this.tokenManagers = this.apps.pipe(
+      map(
+        (apps) => apps.filter((app) => app.appId === tokenManagerID)
+      ),
+      publishReplay(1)
+    )
+    this.tokenManagers.connect()
+  }
+
+  /**
+   * Initialise members observable.
+   *
+   * @return {void}
+   */
+  initMembers () {
+    this.members = new BehaviorSubject({}).pipe(
+      scan(
+        (membersInfo, { address, isMember }) =>
+          Object.assign(membersInfo, { [address]: isMember })
+      ),
+      publishReplay(1)
+    )
+    this.members.connect()
+  }
+
+  /**
+   * Set membership for a specific user.
+   *
+   * @param {string} address The address of the user
+   * @param {string} isMember Whether the address is a member
+   * @return {void}
+   */
+  setMembership (address, isMember) {
+    this.members.next({
+      address,
+      isMember
+    })
+  }
+
+  /**
+   * Initialize the forwardedActions observable
+   *
+   * @return {void}
+   */
+  initForwardedActions () {
+    this.forwardedActions = new BehaviorSubject({}).pipe(
+      scan(
+        (actions, { currentApp, actionId, evmScript = '', state = 0 }) => {
+          const updateIndex = actions.findIndex(
+            action => action.currentApp === currentApp && action.actionId === actionId
+          )
+          if (updateIndex === -1) {
+            // set the last address as the target of the forwarded action
+            const target = evmScript ? this.decodeTransactionPath(evmScript).pop().to : ''
```

Comment:
Not 100% sure this works as intended yet; I think `decodeTransactionPath()` only does one layer of EVMScript "unwrapping" (we will soon be adding in multiple layers).

Right now, if you have a flow like `Token Manager -> Voting -> Finance`, I think this would only get you the Voting app.